backend/tests/test_rocm_download.py

```
@@ -0,0 +1,203 @@
+    async def head(self, url: str):
+        return self._responses.get(url, FakeResponse(status_code=404))
+
+    async def get(self, url: str):
+        return self._responses.get(url, FakeResponse(status_code=404))
+
+    def stream(self, method: str, url: str):
+        resp = self._responses.get(url, FakeResponse(status_code=404))
+        resp.raise_for_status()
+
+        class _Streamer:
+            async def __aenter__(s):
```

Ruff (0.15.10)

[warning] First argument of a method should be named `self`

(N805)